builtins: implement ST_SetSRID

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/49032.
Resolves https://github.com/cockroachdb/cockroach/issues/48403.

Release note (sql change): Implemented ST_SetSRID on geometry and geography
types.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49222)
<!-- Reviewable:end -->

otan committed:
builtins: implement ST_SetSRID

Release note (sql change): Implemented ST_SetSRID on geometry and geography
types.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49222#-:-M7jCxOS8xk6h8JOziG4:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola
